crates/uv/src/commands/tool/mod.rs

```diff
@@ -16,44 +17,87 @@ pub(crate) mod upgrade;
 
 /// A request to run or install a tool (e.g., `uvx ruff@latest`).
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub(crate) struct ToolRequest<'a> {
-    /// The executable name (e.g., `ruff`), if specified explicitly.
-    pub(crate) executable: Option<&'a str>,
-    /// The target to install or run (e.g., `ruff@latest` or `ruff==0.6.0`).
-    pub(crate) target: Target<'a>,
+pub(crate) enum ToolRequest<'a> {
+    // Running the interpreter directly e.g. `uvx python` or `uvx pypy@3.8`
+    Python(PythonRequest),
+    // Running a Python package
+    Package {
+        /// The executable name (e.g., `ruff`), if the target was given via --from.
+        executable: Option<&'a str>,
+        /// The target to install or run (e.g., `ruff@latest` or `ruff==0.6.0`).
+        target: Target<'a>,
+    },
 }
 
 impl<'a> ToolRequest<'a> {
     /// Parse a tool request into an executable name and a target.
     pub(crate) fn parse(command: &'a str, from: Option<&'a str>) -> Self {
+        // If --from is not used, then first try parsing the command as a PythonRequest and see
+        // what we get.
+        if from.is_none() {
+            // As a special case, allow "python" and (on Windows only) "pythonw" here, which
+            // normally aren't valid as PythonRequests (that is, as values of the -p/--python
+            // flag). This is partly for compatibility with older behavior, and partly to allow
+            // scripts to use --python everywhere if they want.
+            if command.eq_ignore_ascii_case("python")
+                || (cfg!(windows) && command.eq_ignore_ascii_case("pythonw"))
+            {
+                return Self::Python(PythonRequest::Default);
+            }
+            // Anything including e.g. `/foo/bar/baz` can be parsed as a PythonRequest, so we're
+            // only interested certain variants here.
+            let python_request = PythonRequest::parse(command);
+            match &python_request {
+                // e.g. `python3`, `python3.12`, or in fact `312`
+                PythonRequest::Version(_) |
```

Comment:
I don't think a bare version (e.g., 312 or 3.12) should work, but I do think `python3` and `python3.12` should work.

Reply:
This suggests that maybe `PythonRequest` isn't quite right since it drops that information? You could add variants like `Interpreter` (i.e., bare `python`) and `InterpreterVersion` (i.e., `python3.13`) (names questionable), but that's a fair amount of work since you'd need to handle them everywhere.